scriptmodules/emulators/mame.sh

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env bash
+
+# This file is part of The RetroPie Project
+#
+# The RetroPie Project is the legal property of its developers, whose names are
+# too numerous to list here. Please refer to the COPYRIGHT.md file distributed with this source.
+#
+# See the LICENSE.md file at the top-level directory of this distribution and
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+
+rp_module_id="mame"
+rp_module_desc="MAME emulator"
+rp_module_help="ROM Extensions: .zip .7z\n\nCopy your MAME roms to either $romdir/mame or\n$romdir/arcade"
+rp_module_licence="GPL2 https://github.com/mamedev/mame/blob/master/COPYING"
+rp_module_section="exp"
+rp_module_flags="!mali !armv6"
+
+function _latest_ver_mame() {
+    wget -qO- https://api.github.com/repos/mamedev/mame/releases/latest | grep -m 1 tag_name | cut -d\" -f4
+}
+
+function _get_binary_name_mame() {
+    # The MAME executable on 64-bit systems is called mame64 instead of mame. Rename it back to mame.
+    if isPlatform "64bit"; then
+        echo 'mame64'
+    else
+        echo 'mame'
+    fi
+}
+
+function depends_mame() {
+    if compareVersions $__gcc_version lt 6.0.0; then
+        md_ret_errors+=("Sorry, you need an OS with gcc 6.0 or newer to compile mame")
+        return 1
+    fi
```

Comment:
I've not tested so sorry if I'm wrong here. lr-mame complains if GCC is lower than 7 - is this tested on GCC 6?

Reply:
Good question. MAME officially only supports GCC 7.2+. When I originally wrote this script I think it was on Raspbian Jesse, but that was also MAME 0.180. On Raspbian Stretch, seems that the last version I compiled was 0.215 and that has GCC 6.3.0.  But according to their docs, GCC 7.2+ has been the supported version since 0.216.

I could run a test on my Stretch install to see if GCC 6.3 still works, but I doubt that it will continue to work in the long run. Plus, I presume more new people will be using RetroPie 4.6 with Raspbian Buster than downloading a previous version. Might as well just update it to 7, like lr-mame. 

Thanks!

Reply:
Updated!

Reply:
Probably best to compare against 7 not 7.2

If GCC -dumpversion returns 7 it will fail the comparison. See recent commits as I've just fixed up some other modules in the same way.

Reply:
Got it. Changed to 7.